javascript/2130-maximum-twin-sum-of-a-linked-list.js

```diff
@@ -0,0 +1,59 @@
+/**
+ * Definition for singly-linked list.
+ * function ListNode(val, next) {
+ *     this.val = (val===undefined ? 0 : val)
+ *     this.next = (next===undefined ? null : next)
+ * }
+ */
+
+/**
+ * Linear Time 
+ * Time O(n) | Space O(1)
+ * @param {ListNode} head
+ * @return {number}
+ */
+const llLength = (head) => {
+  let count = 0;
+  while (head) {
+    head = head.next;
+    count++;
+  }
+  return count;
+};
+
+const reverseLL = (head, len) => {
+  let count = 0;
+  let temp = null;
+  while (count < len) {
+    const next = head.next;
+    head.next = temp;
+    temp = head;
+    head = next;
+    count++;
+  }
+  return temp;
+};
+
+var pairSum = function (head) {
```

Comment:
Suggestion: move your main function to the top of file with complexity comments. And abstract your main function

```js
const length = getLength();
const leftPointer = getLeft();
const rightPointer = getRight();

return getMax();
```

Reply:
I have moved the main function to the top of the file with time complexity comments. What do you mean by "abstract your main function"?

Reply:
The suggestion I posted is an example. The purpose is to back down the algorithm. into modular algorithms. Linked List has common patterns to get the mid, reverse, find cycle, etc. One or more of these patterns are always used in solving problems and abstracting your main function to invoke the patterns.

Reply:
Haven't I modularized the code by writing _llLength_ and _reverseLL_ functions? What else would have me modularized? Any suggestion would be helpful.

Reply:
You can extract out 

const rightPointer = getRightPointer()
const leftPointer = reverseLL(head, mid);
const max = getMax();

Reply:
@aakhtar3, I did modularized it as suggested. Please check. Here's submission [link](https://leetcode.com/problems/maximum-twin-sum-of-a-linked-list/submissions/1042826145/)